sample_project/Source/sample_project/Measure/Measure.cpp

```diff
@@ -0,0 +1,255 @@
+			FVector start = FeaturePoints[i - 1]->GetActorLocation();
+
+			tangent = end - start;
+			tangent.Normalize();
+			tangent = tangent * 100;
+
+			SplineMesh->SetStartAndEnd(start, tangent, end, tangent);
+			SplineMesh->SetStartScale(RouteCueScale);
+			SplineMesh->SetEndScale(RouteCueScale);
+			SplineMesh->SetStaticMesh(RouteMesh);
+			SplineMeshComponents.AddHead(SplineMesh);
+		}
+	}
+}
+
+// Do a line trace from high above to update the elevation info of feature points
+void AMeasure::SetElevation(AActor* stop)
+{
+	FVector position = stop->GetActorLocation();
+	FVector raycastStart = FVector(position.X, position.Y, position.Z + TraceLength);
+	FHitResult hitInfo;
+	float elevationOffset = 200.0f;
+
+	bool bTraceSuccess = GetWorld()->LineTraceSingleByChannel(hitInfo, raycastStart, position + TraceLength * FVector3d::DownVector, ECC_Visibility,
+															  FCollisionQueryParams());
+	position.Z = bTraceSuccess ? hitInfo.ImpactPoint.Z + elevationOffset : 0.0f;
+
+	stop->SetActorLocation(position);
+}
+
+void AMeasure::ClearLine()
+{
+	if (!SplineMeshComponents.IsEmpty())
+	{
+		for (auto point : SplineMeshComponents)
+		{
+			point->UnregisterComponent();
+			point->DestroyComponent();
+		}
+
+		for (auto stop : FeaturePoints)
+		{
+			stop->Destroy();
+		}
+
+		for (auto stop : Stops)
+		{
+			stop->Destroy();
+		}
+
+		SplineMeshComponents.Empty();
+		FeaturePoints.Empty();
+		Stops.Empty();
+		GeodeticDistance = 0;
+		GeodeticDistanceText = FString::Printf(TEXT("Distance: %f %s"), GeodeticDistance, *UnitText);
+		UIWidget->ProcessEvent(WidgetFunction, &GeodeticDistanceText);
+	}
+}
+void AMeasure::UnitChanged()
+{
+	if (UnitDropdown->GetSelectedOption() == "Meters")
+	{
+		Unit = UArcGISLinearUnit::CreateArcGISLinearUnit(EArcGISLinearUnitId::Meters);
+		GeodeticDistance = ConvertUnits(GeodeticDistance, CurrentUnit, meters);
+		CurrentUnit = meters;
+		UnitText = " m";
+	}
+	else if (UnitDropdown->GetSelectedOption() == "Kilometers")
+	{
+		Unit = UArcGISLinearUnit::CreateArcGISLinearUnit(EArcGISLinearUnitId::Kilometers);
+		GeodeticDistance = ConvertUnits(GeodeticDistance, CurrentUnit, kilometers);
+		CurrentUnit = kilometers;
+		UnitText = " km";
+	}
+	else if (UnitDropdown->GetSelectedOption() == "Miles")
+	{
+		Unit = UArcGISLinearUnit::CreateArcGISLinearUnit(EArcGISLinearUnitId::Miles);
+		GeodeticDistance = ConvertUnits(GeodeticDistance, CurrentUnit, miles);
+		CurrentUnit = miles;
+		UnitText = " mi";
+	}
+	else if (UnitDropdown->GetSelectedOption() == "Feet")
+	{
+		Unit = UArcGISLinearUnit::CreateArcGISLinearUnit(EArcGISLinearUnitId::Feet);
+		GeodeticDistance = ConvertUnits(GeodeticDistance, CurrentUnit, feet);
+		CurrentUnit = feet;
+		UnitText = " ft";
+	}
+
+	GeodeticDistanceText = FString::Printf(TEXT("Distance: %f %s"), round(GeodeticDistance * 1000.0) / 1000.0, *UnitText);
+	UIWidget->ProcessEvent(WidgetFunction, &GeodeticDistanceText);
+}
+double AMeasure::ConvertUnits(double value, UnitType from, UnitType to)
```

Comment:
Seems format is not being applied

```suggestion
}

double AMeasure::ConvertUnits(double value, UnitType from, UnitType to)
```

Reply:
We don't need this function. UArcGISLinearUnit has functions on it already to convert to other Units

double ConvertTo(UArcGISLinearUnit* toUnit, double value) const;